File: TheMetz/Services/TeamMemberService.cs
using Microsoft.VisualStudio.Services.WebApi;
using Microsoft.TeamFoundation.Core.WebApi;
using Microsoft.TeamFoundation.WorkItemTracking.WebApi;
using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;

namespace TheMetz.Services
{
    public interface ITeamMemberService
    {
        public Task<List<TeamMember>?> GetCustomerOptimizationTeamMembers();
    }
    
    internal class TeamMemberService : ITeamMemberService
    {
        private readonly VssConnection _connection;
        
        public TeamMemberService(VssConnection connection)
        {
            _connection = connection;
        }
        
        public async Task<List<TeamMember>?> GetCustomerOptimizationTeamMembers()
        {
            var teamClient = await _connection.GetClientAsync<TeamHttpClient>();
            var test = await _connection.GetClientAsync<WorkItemTrackingHttpClient>();
                
            // Retrieve the team members
            List<TeamMember>? members = await teamClient.GetTeamMembersWithExtendedPropertiesAsync(
                projectId: "Marketplace",
                teamId: "Customer Optimization"
            );
            
            // var bizopsTeamMembers = await teamClient.GetTeamMembersWithExtendedPropertiesAsync(
            //     projectId: "Marketplace",
            //     teamId: "Business Optimization"
            // );
            //
            // members.Add(bizopsTeamMembers.First(t => t.Identity.DisplayName == "Brandon George" ));
            // members.Add(bizopsTeamMembers.First(t => t.Identity.DisplayName == "Phil Gathany" ));
            
            return members;
        }
    }
}

File: TheMetz.FSharp/TeamMemberService.fs
namespace TheMetz.FSharp

open System.Threading.Tasks
open Microsoft.TeamFoundation.Core.WebApi
open Microsoft.VisualStudio.Services.WebApi
open TheMetz.Interfaces

type TeamMemberService(connection: VssConnection) =
    interface ITeamMemberService with
        member _.GetCustomerOptimizationTeamMembers() : Task<System.Collections.Generic.List<TeamMember>> =
            task {
                let! teamClient = connection.GetClientAsync<TeamHttpClient>();
                    
                let! members = teamClient.GetTeamMembersWithExtendedPropertiesAsync(
                    projectId = "Marketplace",
                    teamId = "Customer Optimization"
                )
                
                return members
            }
